types/mongodb/index.d.ts

```diff
@@ -2351,11 +2352,61 @@ export interface GridFSBucketWriteStreamOptions extends WriteConcern {
     disableMD5?: boolean;
 }
 
+type Arguments<T> = [T] extends [(...args: infer U) => any]
```

Comment:
Seems like this does the same thing as the built-in type [`Paramters`](https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterst)?

Reply:
Similar, but not entirely. This one also allows you to define arguments as a tuple instead of a function. TBH I mainly just kept what the TypedEventEmitter used because I wanted the emitted events to have good types and I figured it was better to follow practices I've seen elsewhere.

I'll add a comment to clarify